uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Form/AbsencePeriod.php

```diff
@@ -0,0 +1,209 @@
+      $session = CRM_Core_Session::singleton();
+      $session->replaceUserContext($url);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDefaultEntity() {
+    return 'AbsencePeriod';
+  }
+
+  /**
+   * Adds all the fields to the form
+   */
+  private function addFields() {
+    // This hidden field is used to get the AbsencePeriod
+    // id when validating the Order number before submitting the form
+    $this->add('hidden', '_id');
+    $this->add(
+      'text',
+      'title',
+      ts('Title'),
+      $this->getDAOFieldAttributes('title'),
+      true
+    );
+    $this->add(
+      'datepicker',
+      'start_date',
+      ts('Start Date'),
+      $this->getDAOFieldAttributes('start_date'),
+      true,
+      ['time' => false]
+    );
+    $this->add(
+      'datepicker',
+      'end_date',
+      ts('End Date'),
+      $this->getDAOFieldAttributes('end_date'),
+      true,
+      ['time' => false]
+    );
+    $this->add(
+      'number',
+      'weight',
+      ts('Order'),
+      $this->getDAOFieldAttributes('weight'),
+      true
+    );
+  }
+
+  /**
+   * A helper method to call the CRM_Core_DAO::getAttribute to get the
+   * fields attributes of the AbsencePeriod BAO
+   *
+   * @param $field - The name of a field of the AbsencePeriod BAO
+   *
+   * @return array - The attributes returned by CRM_Core_DAO::getAttribute
+   */
+  private function getDAOFieldAttributes($field) {
+    $dao = 'CRM_HRLeaveAndAbsences_DAO_AbsencePeriod';
+    return CRM_Core_DAO::getAttribute($dao, $field);
+  }
+
+  /**
+   * Add validation rules for the fields on this form.
+   */
+  private function addFieldsRules() {
+    $this->addRule('weight', ts('The Order should be a positive number'), 'positiveInteger');
+    $this->addFormRule([$this, 'formRules']);
+  }
+
+  /**
+   * Execute validations concerning all the form fields.
+   *
+   * Different from regular field rules, Form Rules can be used for validations
+   * that depends on multiple fields,(for example, if field X is not empty,
+   * then Y is required). Here, we use it to validate the dates, because start
+   * date can't be greater or equal to end date.
+   *
+   * @param array $values An array containing all the form's fields values
+   *
+   * @return array|bool Returns true if form is valid. Otherwise, an
+   *                    array containing all the validation errors is returned.
+   */
+  public function formRules($values) {
+    $errors = [];
+    $this->validatePeriodDates($values, $errors);
+
+    return empty($errors) ? true : $errors;
+  }
+
+  /**
+   * Validates if the Start Date is less than the End Date
+   *
+   * @param array $values An array containing all the form's fields values
+   * @param array $errors A reference to the errors array where errors will be
+   *                      added if dates are invalid
+   */
+  private function validatePeriodDates($values, &$errors) {
+    $startDate = strtotime($values['start_date']);
```

Comment:
you should check the dates if they are valid dates and in valid format too  , and not just if the start date is less than end date ... since there is no guarantee on the data coming from the user ...